OpenSim/Simulation/Model/KneeLigament.h

```diff
@@ -0,0 +1,237 @@
+\f]
+
+\f[
+f(e) = \frac{1}{4} \cdot \frac{e^2}{e_l}, e \leq 2 \cdot e_l
+\f]
+
+\f[
+f(e) = k \cdot (e - e_l), e > 2 \cdot e_l
+\f]
+
+where, \f$ k \f$ is the stiffness of the ligament and \f$ e_l \f$ is the limit 
+at which the ligament moves from the non-linear region to linear region.
+
+Sometime we define the reference length \f$ l_r \f$ and the reference strain
+\f$ e_r \f$, so the rest length is given by:
+
+\f[
+l_0 = l_r / (e_r + 1)
+\f]
+
+*/
+class OSIMSIMULATION_API KneeLigament : public Force {
+OpenSim_DECLARE_CONCRETE_OBJECT(KneeLigament, Force);
+public:
+//=============================================================================
+// PROPERTIES
+//=============================================================================
+    /** @name Property declarations
+    These are the serializable properties associated with this class. **/
+    /**@{**/    
+	OpenSim_DECLARE_UNNAMED_PROPERTY(GeometryPath, 
+        "the set of points defining the path of the ligament");
+    OpenSim_DECLARE_PROPERTY(resting_length, double,
+        "resting length of the ligament");
+	OpenSim_DECLARE_PROPERTY(epsilon_length, double,
+		"length were the curve changes from non-linear to linear region");
+	OpenSim_DECLARE_PROPERTY(ligament_stiffness, double,
+		"stiffness parameter of the ligament");
+    /**@}**/
+
+
+//==============================================================================
+// PUBLIC METHODS
+//==============================================================================
+    KneeLigament();
+
+    // Uses default (compiler-generated) destructor, copy constructor, and copy
+    // assignment operator.
+
+    //--------------------------------------------------------------------------
+    // GET
+    //--------------------------------------------------------------------------
+    // Properties
+    const GeometryPath& getGeometryPath() const 
+    {   return get_GeometryPath(); }
+    GeometryPath& updGeometryPath() 
+    {   return upd_GeometryPath(); }
+    virtual bool hasGeometryPath() const { return true;};
+    virtual double getLength(const SimTK::State& s) const;
+    virtual double getRestingLength() const 
+    {   return get_resting_length(); }
+    virtual bool setRestingLength(double aRestingLength);
+	
+	const double getEpsilonLength() const
+	{
+		return get_epsilon_length();
+	}
+
+	bool setEpsilonLength(double e_l)
+	{
+		set_epsilon_length(e_l);
+		return true;
+	}
+
+	const double getLigamentStiffness() const
+	{
+		return get_ligament_stiffness();
+	}
+
+	bool setLigamentStiffnessg(double k)
+	{
+		set_ligament_stiffness(k);
+		return true;
+	}
+
+    // computed variables
+    const double& getTension(const SimTK::State& s) const;
+
+    //--------------------------------------------------------------------------
+    // COMPUTATIONS
+    //--------------------------------------------------------------------------
+    virtual double computeMomentArm(const SimTK::State& s, Coordinate& aCoord) const;
```

Comment:
No need to re-declare virtual methods of the base as `virtual`. Instead use the `override` keyword to indicate overridden methods of a base class. E.g:
`void computeForce(const SimTK::State& s, ... ) const override;`